diff --git a/proftpd/proftpd.conf b/proftpd/proftpd.conf

To set password run this: sudo passwd ftp

--- a/proftpd/proftpd.conf
+++ b/proftpd/proftpd.conf
@@ -134,16 +134,17 @@ AdminControlsEngine off
 
 # A basic anonymous configuration, no upload directories.
 
-# <Anonymous ~ftp>
-#   User				ftp
-#   Group				nogroup
+<Anonymous ~ftp>
+  AnonRequirePassword		on
+  User				ftp
+  Group				nogroup
 #   # We want clients to be able to login with "anonymous" as well as "ftp"
 #   UserAlias			anonymous ftp
 #   # Cosmetic changes, all files belongs to ftp user
 #   DirFakeUser	on ftp
 #   DirFakeGroup on ftp
 # 
-#   RequireValidShell		off
+  RequireValidShell		off
 # 
 #   # Limit the maximum number of anonymous logins
 #   MaxClients			10
@@ -173,4 +174,4 @@ AdminControlsEngine off
 #   #            </Limit>
 #   # </Directory>
 # 
-# </Anonymous>
+</Anonymous>
