--- app_voicemail.c.orig	2011-10-27 14:43:14.000000000 +0800
+++ app_voicemail.c	2011-10-27 14:41:51.000000000 +0800
@@ -7107,10 +7107,10 @@
 	vms->starting = 0;
 	make_file(vms->fn, sizeof(vms->fn), vms->curdir, vms->curmsg);
 	adsi_message(chan, vms);
-	if (!vms->curmsg)
-		res = wait_file2(chan, vms, "vm-first");	/* "First" */
-	else if (vms->curmsg == vms->lastmsg)
-		res = wait_file2(chan, vms, "vm-last");		/* "last" */
+	//if (!vms->curmsg)
+	//	res = wait_file2(chan, vms, "vm-first");	/* "First" */
+	//else if (vms->curmsg == vms->lastmsg)
+	//	res = wait_file2(chan, vms, "vm-last");		/* "last" */
 
 	snprintf(filename, sizeof(filename), "%s.txt", vms->fn);
 	RETRIEVE(vms->curdir, vms->curmsg, vmu->mailbox, vmu->context);
@@ -7170,11 +7170,11 @@
 			} else { /* DEFAULT syntax */
 				res = wait_file2(chan, vms, "vm-message");
 			}
-			if (vms->curmsg && (vms->curmsg != vms->lastmsg)) {
+			//if (vms->curmsg && (vms->curmsg != vms->lastmsg)) {
 				if (!res) {
 					res = ast_say_number(chan, vms->curmsg + 1, AST_DIGIT_ANY, chan->language, NULL);
 				}
-			}
+			//}
 		}
 	}
 
@@ -9063,7 +9063,10 @@
 	}
 	
 	/* Authenticate them and get their mailbox/password */
-	
+
+	if (skipuser)
+		valid++;
+
 	while (!valid && (logretries < max_logins)) {
 		/* Prompt for, and read in the username */
 		if (!skipuser && ast_readstring(chan, mailbox, mailbox_size - 1, 2000, 10000, "#") < 0) {
@@ -9766,16 +9769,16 @@
 			break;
 		case '*': /* Help */
 			if (!vms.starting) {
-				cmd = ast_play_and_wait(chan, "vm-onefor");
-				if (!strncasecmp(chan->language, "he", 2)) {
-					cmd = ast_play_and_wait(chan, "vm-for");
-				}
-				if (!cmd)
-					cmd = vm_play_folder_name(chan, vms.vmbox);
-				if (!cmd)
-					cmd = ast_play_and_wait(chan, "vm-opts");
-				if (!cmd)
-					cmd = vm_instructions(chan, vmu, &vms, 1, in_urgent);
+				//cmd = ast_play_and_wait(chan, "vm-onefor");
+				//if (!strncasecmp(chan->language, "he", 2)) {
+				//	cmd = ast_play_and_wait(chan, "vm-for");
+				//}
+				//if (!cmd)
+				//	cmd = vm_play_folder_name(chan, vms.vmbox);
+				//if (!cmd)
+				//	cmd = ast_play_and_wait(chan, "vm-opts");
+				//if (!cmd)
+					cmd = vm_instructions(chan, vmu, &vms, 0, in_urgent);
 			} else
 				cmd = 0;
 			break;
